error[E0506]: cannot assign to `x` because it is borrowed (Ast)
  --> $DIR/maybe-initialized-drop-implicit-fragment-drop.rs:31:5
   |
27 |     let wrap = Wrap { p: &mut x };
   |                               - borrow of `x` occurs here
...
31 |     x = 1; //~ ERROR because of Foo.a's implicit dtor
   |     ^^^^^ assignment to borrowed `x` occurs here

error[E0506]: cannot assign to `x` because it is borrowed (Mir)
  --> $DIR/maybe-initialized-drop-implicit-fragment-drop.rs:31:5
   |
27 |     let wrap = Wrap { p: &mut x };
   |                          ------ borrow of `x` occurs here
...
31 |     x = 1; //~ ERROR because of Foo.a's implicit dtor
   |     ^^^^^ assignment to borrowed `x` occurs here

error: aborting due to 2 previous errors

