error: some polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:12:1
   |
LL | / fn dependency_because_offset_depends_on_T<T>(parameter: &OffsetDependent<T>) -> u32 {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     parameter.count
   | |     --------------- depends on `OffsetOf(OffsetDependent<T>, field[1])`
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:17:1
   |
LL | / fn dependency_because_offset_depends_on_T_sized<T: ?Sized>(parameter: &OffsetDependent<&T>) -> u32 {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     parameter.count
   | |     --------------- depends on `OffsetOf(OffsetDependent<&T>, field[1])`
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:22:1
   |
LL | / fn no_dependency_because_offset<T>(parameter: &OffsetDependent<&T>) -> u32 {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // Correct: size/alignment of `T` known.
LL | |     parameter.count
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:29:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: aborting due to 4 previous errors

