warning: no polymorphic dependencies found
  --> $DIR/polymorphize.rs:8:1
   |
LL | / fn no_use<T>() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize.rs:12:1
   |
LL | / fn no_dependency_because_pointer<T>(t: &T) -> &T {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     t
LL | | }
   | |_^

warning: some polymorphic dependencies found
  --> $DIR/polymorphize.rs:17:1
   |
LL | / fn dependency_because_unsized_pointer<T: ?Sized>(t: &T) -> &T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     t
   | |     - depends on `SizeAlignment(&T)`
LL | | }
   | |_^

warning: some polymorphic dependencies found
  --> $DIR/polymorphize.rs:22:1
   |
LL | / fn depend_size_alignment<T: Copy>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     *t
   | |     -- depends on `SizeAlignment(T)`
LL | | }
   | |_^

warning: some polymorphic dependencies found
  --> $DIR/polymorphize.rs:27:1
   |
LL |       *t
   |       -- depends on `SizeAlignment(T)`
...
LL | / fn depend_size_alignment_indirect<T: Copy>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     depend_size_alignment(t)
   | |     ------------------------ invokes `DefId(0/0:6 ~ polymorphize[317d]::depend_size_alignment[0])` with substitutions `[T]`
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize.rs:32:1
   |
LL | / fn no_dependency_indirect<T: Copy>(t: &T) -> u32 {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     depend_size_alignment(&22)
   | |     -------------------------- invokes `DefId(0/0:6 ~ polymorphize[317d]::depend_size_alignment[0])` with substitutions `[u32]`
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize.rs:37:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

note: Monomorphized items: 9

note: Monomorphized size : 32

note: New total items    : 9 (100%)

note: New estimated size : 32 (100%)

