warning: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:12:1
   |
LL | / fn no_dependency_because_pointer<T>(t: &T) -> &T {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     t
LL | | }
   | |_^

warning: some polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:17:1
   |
LL | / fn dependency_because_unsized_pointer<T: ?Sized>(t: &T) -> &T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     t
   | |     - depends on `SizeAlignment(&T)`
LL | | }
   | |_^

warning: some polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:22:1
   |
LL | / fn dependency_because_embed_ref_unsized<T: ?Sized>(t: &T) -> EmbedRef<'_, T> {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     EmbedRef { t }
   | |     --------------
   | |     |          |
   | |     |          depends on `SizeAlignment(&T)`
   | |     depends on `OffsetOf(EmbedRef<T>, field[0])`
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:27:1
   |
LL | / fn dependency_because_embed_ref_sized<T>(t: &T) -> EmbedRef<'_, T> {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // Here, the size of `EmbedRef` is known up front.
LL | |     EmbedRef { t }
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:34:1
   |
LL | / fn no_dependency_because_unsized_pointer_indirect<T>(t: &T) {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |
LL | |     // Although `dependency_because_unsized_pointer` has a dependency
...  |
LL | |     dependency_because_unsized_pointer::<T>(t);
   | |     ------------------------------------------ invokes `DefId(0/0:5 ~ polymorphize_sized[317d]::dependency_because_unsized_pointer[0])` with substitutions `[T]`
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:43:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |
LL | |     no_dependency_because_pointer::<u32>(&22);
   | |     ----------------------------------------- invokes `DefId(0/0:4 ~ polymorphize_sized[317d]::no_dependency_because_pointer[0])` with substitutions `[u32]`
LL | |     no_dependency_because_pointer::<u16>(&22);
   | |     ----------------------------------------- invokes `DefId(0/0:4 ~ polymorphize_sized[317d]::no_dependency_because_pointer[0])` with substitutions `[u16]`
LL | |
LL | |     dependency_because_unsized_pointer::<u32>(&22);
   | |     ---------------------------------------------- invokes `DefId(0/0:5 ~ polymorphize_sized[317d]::dependency_because_unsized_pointer[0])` with substitutions `[u32]`
LL | |     dependency_because_unsized_pointer::<u16>(&22);
   | |     ---------------------------------------------- invokes `DefId(0/0:5 ~ polymorphize_sized[317d]::dependency_because_unsized_pointer[0])` with substitutions `[u16]`
LL | |
LL | |     no_dependency_because_unsized_pointer_indirect::<u32>(&22);
   | |     ---------------------------------------------------------- invokes `DefId(0/0:8 ~ polymorphize_sized[317d]::no_dependency_because_unsized_pointer_indirect[0])` with substitutions `[u32]`
LL | |     no_dependency_because_unsized_pointer_indirect::<u16>(&22);
   | |     ---------------------------------------------------------- invokes `DefId(0/0:8 ~ polymorphize_sized[317d]::no_dependency_because_unsized_pointer_indirect[0])` with substitutions `[u16]`
LL | | }
   | |_^

note: Monomorphized items: 15

note: Monomorphized size : 78

note: New total items    : 13 ( 87%)

note: New estimated size : 74 ( 95%)

