error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:11:1
   |
LL | / fn no_dependency_because_pointer<T>(t: &T) -> &T {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     t
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:16:1
   |
LL | / fn dependency_because_unsized_pointer<T: ?Sized>(t: &T) -> &T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     t
   | |     - depends on `SizeAlignment(&T)`
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:21:1
   |
LL | / fn dependency_because_embed_ref_unsized<T: ?Sized>(t: &T) -> EmbedRef<'_, T> {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     EmbedRef { t }
   | |     --------------
   | |     |          |
   | |     |          depends on `SizeAlignment(&T)`
   | |     depends on `SizeAlignment(T)`
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:26:1
   |
LL | / fn dependency_because_embed_ref_sized<T>(t: &T) -> EmbedRef<'_, T> {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // Here, the size of `EmbedRef` is known up front.
LL | |     EmbedRef { t }
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:33:1
   |
LL | / fn no_dependency_because_unsized_pointer_indirect<T>(t: &T) {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |
LL | |     // Although `dependency_because_unsized_pointer` has a dependency
...  |
LL | |     dependency_because_unsized_pointer::<T>(t);
   | |     ------------------------------------------ invokes `DefId(0/0:5 ~ polymorphize_sized[317d]::dependency_because_unsized_pointer[0])` with substitutions `[T]`
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:42:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: aborting due to 6 previous errors

