error: some polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:7:1
   |
LL | / fn depend_trait_dispatch<T: Clone>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     t.clone()
   | |     --------- depends on type `T/#0` for `TraitMethod`
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:12:1
   |
LL |       t.clone()
   |       --------- depends on type `T/#0` for `TraitMethod`
...
LL | / fn depend_trait_dispatch_indirect<T: Clone>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     depend_trait_dispatch(t)
   | |     ------------------------ invokes `DefId(0/0:3 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch[0])` with substitutions `[T]`
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:17:1
   |
LL | / fn depend_trait_dispatch_ref_indirect<T: Clone>(t: &T) -> &T {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // FIXME this should be a dependency, but right now we see that we
...  |
LL | |     depend_trait_dispatch::<&T>(&t)
   | |     ------------------------------- invokes `DefId(0/0:3 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch[0])` with substitutions `[&T]`
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:28:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: aborting due to 4 previous errors

