error: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:12:1
   |
LL | / fn dependency_because_offset_depends_on_T<T>(parameter: &OffsetDependent<T>) -> u32 {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // FIXME -- the offset of the `count` field depends on size/alignment of `T`
LL | |     parameter.count
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:19:1
   |
LL | / fn dependency_because_offset_depends_on_T_sized<T: ?Sized>(parameter: &OffsetDependent<&T>) -> u32 {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // FIXME -- the offset of the `count` field depends on whether `T` is sized
LL | |     parameter.count
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:26:1
   |
LL | / fn no_dependency_because_offset<T>(parameter: &OffsetDependent<&T>) -> u32 {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // Correct: size/alignment of `T` known.
LL | |     parameter.count
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:33:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: aborting due to 4 previous errors

