warning: some polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:8:1
   |
LL | / fn depend_trait_dispatch<T: Clone>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     t.clone()
   | |     --------- depends on `TraitMethod(DefId(2/0:1278 ~ core[8468]::clone[0]::Clone[0]::clone[0]), [T])`
LL | | }
   | |_^

warning: some polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:13:1
   |
LL |       t.clone()
   |       --------- depends on `TraitMethod(DefId(2/0:1278 ~ core[8468]::clone[0]::Clone[0]::clone[0]), [T])`
...
LL | / fn depend_trait_dispatch_indirect<T: Clone>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     depend_trait_dispatch(t)
   | |     ------------------------ invokes `DefId(0/0:3 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch[0])` with substitutions `[T]`
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:18:1
   |
LL |       t.clone()
   |       --------- invokes `DefId(2/0:1292 ~ core[8468]::clone[0]::impls[0]::{{impl}}[3]::clone[0])` with substitutions `[ReErased, T]`
...
LL | / fn depend_trait_dispatch_ref_indirect<T: Clone>(t: &T) -> &T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     depend_trait_dispatch::<&T>(&t)
   | |     ------------------------------- invokes `DefId(0/0:3 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch[0])` with substitutions `[&T]`
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:23:1
   |
LL |       t.clone()
   |       ---------
   |       |
   |       invokes `DefId(2/0:17671 ~ core[8468]::clone[0]::impls[0]::{{impl}}[7]::clone[0])` with substitutions `[]`
   |       invokes `DefId(2/0:17669 ~ core[8468]::clone[0]::impls[0]::{{impl}}[6]::clone[0])` with substitutions `[]`
...
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |
LL | |     // Invoke each function so that they are considered by the collector.
LL | |     depend_trait_dispatch::<u32>(&3);
   | |     -------------------------------- invokes `DefId(0/0:3 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch[0])` with substitutions `[u32]`
LL | |     depend_trait_dispatch::<u16>(&3);
   | |     -------------------------------- invokes `DefId(0/0:3 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch[0])` with substitutions `[u16]`
LL | |
LL | |     depend_trait_dispatch_indirect::<u32>(&3);
   | |     ----------------------------------------- invokes `DefId(0/0:4 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch_indirect[0])` with substitutions `[u32]`
LL | |     depend_trait_dispatch_indirect::<u16>(&3);
   | |     ----------------------------------------- invokes `DefId(0/0:4 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch_indirect[0])` with substitutions `[u16]`
LL | |
LL | |     depend_trait_dispatch_ref_indirect::<u32>(&3);
   | |     --------------------------------------------- invokes `DefId(0/0:5 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch_ref_indirect[0])` with substitutions `[u32]`
LL | |     depend_trait_dispatch_ref_indirect::<u16>(&3);
   | |     --------------------------------------------- invokes `DefId(0/0:5 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch_ref_indirect[0])` with substitutions `[u16]`
LL | | }
   | |_^

note: Monomorphized items: 21

note: Monomorphized size : 108

note: New total items    : 20 ( 95%)

note: New estimated size : 99 ( 92%)

