warning: some polymorphic dependencies found
  --> $DIR/polymorphize_drop.rs:13:1
   |
LL | / fn dependency_because_drop_T<T>(data: T) {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | | }
   | | ^
   | | |
   | |_depends on `Drop(T)`
   | 

warning: some polymorphic dependencies found
  --> $DIR/polymorphize_drop.rs:17:1
   |
LL | / fn dependency_because_drop_box_T<T>(data: Box<T>) {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | | }
   | | ^
   | | |
   | |_depends on `Drop(std::boxed::Box<T>)`
   | 

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_drop.rs:21:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

note: Monomorphized items: 9

note: Monomorphized size : 32

note: New total items    : 9 (100%)

note: New estimated size : 32 (100%)

