error: no polymorphic dependencies found
  --> $DIR/polymorphize.rs:7:1
   |
LL | / fn no_use<T>() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize.rs:11:1
   |
LL | / fn no_dependency_because_pointer<T>(t: &T) -> &T {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     t
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize.rs:16:1
   |
LL | / fn dependency_because_unsized_pointer<T: ?Sized>(t: &T) -> &T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     t
   | |     - depends on type `T/#0` for `SizeAlignment`
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize.rs:21:1
   |
LL | / fn depend_size_alignment<T: Copy>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     *t
   | |     -- depends on type `T/#0` for `SizeAlignment`
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize.rs:26:1
   |
LL |       *t
   |       -- depends on type `T/#0` for `SizeAlignment`
...
LL | / fn depend_size_alignment_indirect<T: Copy>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     depend_size_alignment(t)
   | |     ------------------------ invokes `DefId(0/0:6 ~ polymorphize[317d]::depend_size_alignment[0])` with substitutions `[T]`
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize.rs:31:1
   |
LL | / fn no_dependency_indirect<T: Copy>(t: &T) -> u32 {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     depend_size_alignment(&22)
   | |     -------------------------- invokes `DefId(0/0:6 ~ polymorphize[317d]::depend_size_alignment[0])` with substitutions `[u32]`
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize.rs:36:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: aborting due to 7 previous errors

