warning: some polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:13:1
   |
LL | / fn dependency_because_offset_depends_on_T<T>(parameter: &OffsetDependent<T>) -> u32 {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     parameter.count
   | |     --------------- depends on `OffsetOf(OffsetDependent<T>, field[1])`
LL | | }
   | |_^

warning: some polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:18:1
   |
LL | / fn dependency_because_offset_depends_on_T_sized<T: ?Sized>(parameter: &OffsetDependent<&T>) -> u32 {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     parameter.count
   | |     --------------- depends on `OffsetOf(OffsetDependent<&T>, field[1])`
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:23:1
   |
LL | / fn no_dependency_because_offset<T>(parameter: &OffsetDependent<&T>) -> u32 {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // Correct: size/alignment of `T` known.
LL | |     parameter.count
LL | | }
   | |_^

warning: some polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:30:1
   |
LL | / fn dependency_because_index_depends_on_T<T>(parameters: &[T]) -> &T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     &parameters[0]
   | |     -------------- depends on `IndexInto([T], _3)`
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:35:1
   |
LL | / fn no_dependency_because_index_depends_on_sized_pointer<'a, T>(parameters: &'a [&'a T]) -> &'a T {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     &parameters[0]
LL | | }
   | |_^

warning: no polymorphic dependencies found
  --> $DIR/polymorphize_place.rs:40:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |
LL | |     // Invoke each function so that they are considered by the collector.
LL | |     no_dependency_because_offset::<u32>(&OffsetDependent { t: &22, count: 2 });
   | |     -------------------------------------------------------------------------- invokes `DefId(0/0:6 ~ polymorphize_place[317d]::no_dependency_because_offset[0])` with substitutions `[u32]`
LL | |     no_dependency_because_offset::<u16>(&OffsetDependent { t: &22, count: 2 });
   | |     -------------------------------------------------------------------------- invokes `DefId(0/0:6 ~ polymorphize_place[317d]::no_dependency_because_offset[0])` with substitutions `[u16]`
LL | |
LL | |     dependency_because_offset_depends_on_T_sized::<u32>(&OffsetDependent { t: &22, count: 2 });
   | |     ------------------------------------------------------------------------------------------ invokes `DefId(0/0:5 ~ polymorphize_place[317d]::dependency_because_offset_depends_on_T_sized[0])` with substitutions `[u32]`
LL | |     dependency_because_offset_depends_on_T_sized::<u16>(&OffsetDependent { t: &22, count: 2 });
   | |     ------------------------------------------------------------------------------------------ invokes `DefId(0/0:5 ~ polymorphize_place[317d]::dependency_because_offset_depends_on_T_sized[0])` with substitutions `[u16]`
LL | |     dependency_because_offset_depends_on_T_sized::<str>(&OffsetDependent { t: "foo", count: 2 });
   | |     -------------------------------------------------------------------------------------------- invokes `DefId(0/0:5 ~ polymorphize_place[317d]::dependency_because_offset_depends_on_T_sized[0])` with substitutions `[str]`
LL | |
LL | |     dependency_because_offset_depends_on_T::<u32>(&OffsetDependent { t: 22, count: 2 });
   | |     ----------------------------------------------------------------------------------- invokes `DefId(0/0:4 ~ polymorphize_place[317d]::dependency_because_offset_depends_on_T[0])` with substitutions `[u32]`
LL | |     dependency_because_offset_depends_on_T::<u16>(&OffsetDependent { t: 22, count: 2 });
   | |     ----------------------------------------------------------------------------------- invokes `DefId(0/0:4 ~ polymorphize_place[317d]::dependency_because_offset_depends_on_T[0])` with substitutions `[u16]`
LL | |
LL | |     dependency_because_index_depends_on_T::<u32>(&[2, 3, 4]);
   | |     -------------------------------------------------------- invokes `DefId(0/0:7 ~ polymorphize_place[317d]::dependency_because_index_depends_on_T[0])` with substitutions `[u32]`
LL | |     dependency_because_index_depends_on_T::<u16>(&[2, 3, 4]);
   | |     -------------------------------------------------------- invokes `DefId(0/0:7 ~ polymorphize_place[317d]::dependency_because_index_depends_on_T[0])` with substitutions `[u16]`
LL | |
LL | |     no_dependency_because_index_depends_on_sized_pointer::<u32>(&[&2, &3, &4]);
   | |     -------------------------------------------------------------------------- invokes `DefId(0/0:8 ~ polymorphize_place[317d]::no_dependency_because_index_depends_on_sized_pointer[0])` with substitutions `[u32]`
LL | |     no_dependency_because_index_depends_on_sized_pointer::<u16>(&[&2, &3, &4]);
   | |     -------------------------------------------------------------------------- invokes `DefId(0/0:8 ~ polymorphize_place[317d]::no_dependency_because_index_depends_on_sized_pointer[0])` with substitutions `[u16]`
LL | | }
   | |_^

note: Detected duplicate of DefId(0/0:6 ~ polymorphize_place[317d]::no_dependency_because_offset[0]) with dependencies []

note: Detected duplicate of DefId(0/0:8 ~ polymorphize_place[317d]::no_dependency_because_index_depends_on_sized_pointer[0]) with dependencies []

note: Monomorphized items: 20

note: Monomorphized size : 149

note: New total items    : 18 ( 90%)

note: New estimated size : 140 ( 94%)

