error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:11:1
   |
LL | / fn no_dependency_because_pointer<T>(t: &T) -> &T {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     t
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:16:1
   |
LL | / fn dependency_because_unsized_pointer<T: ?Sized>(t: &T) -> &T {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // FIXME-- we do depend on `T`, because size of `&T` depends on knowning whether
LL | |     // `T` is sized or not
LL | |     t
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:24:1
   |
LL | / fn dependency_because_embed_ref_unsized<T: ?Sized>(t: &T) -> EmbedRef<'_, T> {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // FIXME-- we do depend on `T`, because size of `&T` depends on knowning whether
LL | |     // `T` is sized or not
LL | |     EmbedRef { t }
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:32:1
   |
LL | / fn dependency_because_embed_ref_sized<T>(t: &T) -> EmbedRef<'_, T> {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     EmbedRef { t }
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_sized.rs:37:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: aborting due to 5 previous errors

