error: no polymorphic dependencies found
  --> $DIR/polymorphize_drop.rs:12:1
   |
LL | / fn dependency_because_drop_T<T>(data: T) {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // We have to drop `data`, and to know what to run, we have to know what `T` is.
LL | |     //
LL | |     // FIXME-- we don't presently see the dependency
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_drop.rs:20:1
   |
LL | / fn dependency_because_drop_box_T<T>(data: Box<T>) {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | |     //
LL | |     // We have to drop `data`, and to know what to run, we have to
...  |
LL | |     // FIXME-- we don't presently see the dependency
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_drop.rs:30:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: aborting due to 3 previous errors

