error: some polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:7:1
   |
LL | / fn depend_trait_dispatch<T: Clone>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     t.clone()
   | |     --------- depends on `TraitMethod(DefId(2/0:1278 ~ core[c653]::clone[0]::Clone[0]::clone[0]), [T])`
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:12:1
   |
LL |       t.clone()
   |       --------- depends on `TraitMethod(DefId(2/0:1278 ~ core[c653]::clone[0]::Clone[0]::clone[0]), [T])`
...
LL | / fn depend_trait_dispatch_indirect<T: Clone>(t: &T) -> T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     depend_trait_dispatch(t)
   | |     ------------------------ invokes `DefId(0/0:3 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch[0])` with substitutions `[T]`
LL | | }
   | |_^

error: some polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:17:1
   |
LL |       t.clone()
   |       ---------
   |       |
   |       depends on `OtherMethod([ReErased, T])`
   |       invokes `DefId(2/0:1292 ~ core[c653]::clone[0]::impls[0]::{{impl}}[3]::clone[0])` with substitutions `[ReErased, T]`
...
LL | / fn depend_trait_dispatch_ref_indirect<T: Clone>(t: &T) -> &T {
LL | |     //~^ ERROR some polymorphic dependencies found
LL | |     depend_trait_dispatch::<&T>(&t)
   | |     ------------------------------- invokes `DefId(0/0:3 ~ polymorphize_trait_dispatch[317d]::depend_trait_dispatch[0])` with substitutions `[&T]`
LL | | }
   | |_^

error: no polymorphic dependencies found
  --> $DIR/polymorphize_trait_dispatch.rs:22:1
   |
LL | / fn main() {
LL | |     //~^ ERROR no polymorphic dependencies found
LL | | }
   | |_^

error: aborting due to 4 previous errors

